### PR TITLE
Feature/tigon archetypes

### PR DESCRIPTION
**Pull request by ParthGandhi**
Created a new unit-test top level module. 

Created a maven-archetype for Tigon. Refer to README for usage.

### CODE REVIEW

File: tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      <groupId>co.cask.tigon</groupId>
+      <artifactId>tigon-unit-test</artifactId>
+      <version>${tigon.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>${slf4j.version}</version>
+      <scope>provided</scope>
```
Comment (ParthGandhi) [inline tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml]: Should not be provided.

**Comment by gokulavasan** [comment]
@ParthGandhi @sreevatsanraman Can you guys take a look and see if it is good. We can merge this if its all good. I checked on standalone and distributed and it seems to work fine.

File: tigon-archetypes/pom.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  ~ the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>tigon</artifactId>
+    <groupId>co.cask.tigon</groupId>
+    <version>0.1.0-SNAPSHOT</version>
```
Comment (sreevatsanraman) [inline tigon-archetypes/pom.xml]: Remove snapshot

File: tigon-archetypes/tigon-app-archetype/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+# Creating
+
+To create a project from the archetype, use this script as an example:
+
+```
+ mvn archetype:generate
+  -DarchetypeGroupId=co.cask.tigon
+  -DarchetypeArtifactId=tigon-app-archetype
+  -DarchetypeVersion=0.1.0-SNAPSHOT
```
Comment (sreevatsanraman) [inline tigon-archetypes/tigon-app-archetype/README.md]: 0.1.0

File: tigon-archetypes/tigon-app-archetype/pom.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  ~ the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>co.cask.tigon</groupId>
+    <artifactId>tigon-archetypes</artifactId>
+    <version>0.1.0-SNAPSHOT</version>
```
Comment (sreevatsanraman) [inline tigon-archetypes/tigon-app-archetype/pom.xml]: 0.1.0

File: tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  <artifactId>${artifactId}</artifactId>
+  <version>${version}</version>
+  <packaging>jar</packaging>
+
+  <name>Tigon Application</name>
+
+  <properties>
+    <app.main.class>${package}.CountRandom</app.main.class>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <tigon.version>0.1.0-SNAPSHOT</tigon.version>
```
Comment (sreevatsanraman) [inline tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml]: 0.1.0

File: tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    <slf4j.version>1.7.5</slf4j.version>
+    <guava.version>13.0.1</guava.version>
+    <junit.version>4.11</junit.version>
+  </properties>
+
+  <repositories>
+    <repository>
+      <id>sonatype</id>
```
Comment (sreevatsanraman) [inline tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml]: We don't need sonatype deps.

File: tigon-archetypes/tigon-app-archetype/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
+# Tigon Application Archetype
+
+This directory contains a Tigon Application Archetype for Tigon.
```
Comment (awholegunch) [inline tigon-archetypes/tigon-app-archetype/README.md]: This directory contains a Maven Application Archetype for Tigon.

**Comment by sreevatsanraman** [comment]
Minor comments. Fix and merg.e

File: tigon-archetypes/pom.xml (hunk 2 above)
Comment (awholegunch) [inline tigon-archetypes/pom.xml]: Isn't there a mvn script to use here?

File: tigon-archetypes/pom.xml [hunk 8]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright © 2014 Cask Data, Inc.
```
Comment (chtyim) [inline tigon-archetypes/pom.xml]: No `~`. You can adjust the copyright settings in IntelliJ (Preferences -> Copyright -> Formatting -> XML -> Use custom formatting options)

File: tigon-archetypes/pom.xml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  <artifactId>tigon-archetypes</artifactId>
+  <packaging>pom</packaging>
+  <name>Tigon Application Archetypes</name>
+
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.apache.rat</groupId>
+          <artifactId>apache-rat-plugin</artifactId>
```
Comment (chtyim) [inline tigon-archetypes/pom.xml]: Why? Shouldn't it in the parent pom?

File: tigon-archetypes/pom.xml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+          <executions>
+            <execution>
+              <id>rat-check</id>
+              <phase>validate</phase>
+              <goals>
+                <goal>check</goal>
+              </goals>
+              <configuration>
+                <excludes>
+                  <exclude>**/target/**</exclude>
```
Comment (chtyim) [inline tigon-archetypes/pom.xml]: Also shouldn't have this no matter where you are configuring rat-check.

File: tigon-archetypes/pom.xml [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+                </excludes>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+  </build>
+
+</project>
```
Comment (chtyim) [inline tigon-archetypes/pom.xml]: newline.

File: tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>co.cask.tigon</groupId>
+      <artifactId>tigon-api</artifactId>
+      <version>${tigon.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>co.cask.tigon</groupId>
+      <artifactId>tigon-sql</artifactId>
```
Comment (chtyim) [inline tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml]: Not sure if you want this as default. No that many people will actually start with using tigon-sql.

File: tigon-sql/pom.xml [hunk 13]
```diff
@@ -107,13 +102,13 @@
     <dependency>
       <groupId>co.cask.tephra</groupId>
       <artifactId>tephra-core</artifactId>
-      <version>0.3.0</version>
+      <version>${tephra.version}</version>
```
Comment (chtyim) [inline tigon-sql/pom.xml]: Isn't that we are using dependency management? In child pom.xml, we should only have groupId, artifactId and scope.

**Comment by ParthGandhi** [comment]
@sreevatsanraman @poornachandra Please review.

File: tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+      <groupId>co.cask.tigon</groupId>
+      <artifactId>tigon-api</artifactId>
+      <version>${tigon.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>co.cask.tigon</groupId>
+      <artifactId>tigon-unit-test</artifactId>
+      <version>${tigon.version}</version>
+      <scope>provided</scope>
```
Comment (sreevatsanraman) [inline tigon-archetypes/tigon-app-archetype/src/main/resources/archetype-resources/pom.xml]: Scope should be test

**Comment by sreevatsanraman** [comment]
one comment - fix and merge.

**Comment by ParthGandhi** [comment]
https://bamboo-prod.continuuity.com/browse/PRODRELALL-UNIT23-1

**Pull request merged**